crypto/asn1/a_bitstr.c

```diff
@@ -114,7 +114,7 @@ ASN1_BIT_STRING *c2i_ASN1_BIT_STRING(ASN1_BIT_STRING **a,
     ret->flags |= (ASN1_STRING_FLAG_BITS_LEFT | i); /* set */
 
     if (len-- > 1) {            /* using one because of the bits left byte */
-        s = OPENSSL_malloc((int)len);
+        s = (unsigned char *)OPENSSL_malloc((int)len);
```

Comment:
Those malloc/realloc casts shouldn't be necessary, since they return "void *", right?

Reply:
@richsalz
With VS2019 at highest warning level, a cast from void* to [anything else] requires an explicit cast.

Reply:
Is there a "one below highest" that doesn't complain about missing cast-from-void* ?  That, and the combination of @kroeckx note makes me think this is quiet for its own sake, not code cleanup.

Reply:
@richsalz 

In VS2019, a void* cast issue is not even a warning.  It's an error that can't be suppressed - there's no way to compile the code as-written.

Reply:
> In VS2019, a void* cast issue is not even a warning. It's an error that can't be suppressed - there's no way to compile the code as-written

That doesn't make sense.  I'm surprised.  Looking at https://docs.microsoft.com/en-us/cpp/cpp/void-cpp?view=vs-2019, it says the  cast is optional in C and required in C++.

Reply:
Assuming that's the case (which I have not confirmed locally)...
Is it the openssl group's stance that openssl can never be compiled as C++, i.e. you explicitly say "we do not support C++ compilation"?

Reply:
Someone currently on the project will have to confirm that OpenSSL itself needs a C, not C++ compiler, although it can be used with C++ code.

Reply:
The public headers do this..
```
#ifdef  __cplusplus
extern "C" {
#endif

# ifdef  __cplusplus
}
# endif
```

Reply:
@richsalz 
Slontis's reply effectively implies that C++ compilation is supposed to be supported.  Otherwise, there's no reason to have those.

Reply:
No, @slontis's note that says OpenSSL headers can be used in C++ applications; not that OpenSSL itself can be compiled with a C++ compiler.

Reply:
@richsalz  - Who can answer the question as to whether OpenSSL explicitly does/does not support C++ compilation?

Reply:
I suppose either @openssl/omc or @openssl/otc can definitively say.

Reply:
OpenSSL code (except for the public headers of course) is NOT supposed to be compiled as C++ code.